From db45750963292c30f0eb84833b522a42fd9a2fe4 Mon Sep 17 00:00:00 2001
From: Charles Hu <yizhe_hu@apple.com>
Date: Thu, 25 Jul 2024 14:52:24 -0700
Subject: [PATCH] putil.cpp SwiftFoundationICU changes

---
 icuSources/common/putil.cpp | 40 -------------------------------------
 1 file changed, 40 deletions(-)

diff --git a/icuSources/common/putil.cpp b/icuSources/common/putil.cpp
index 89216be..f318f56 100644
--- a/icuSources/common/putil.cpp
+++ b/icuSources/common/putil.cpp
@@ -1426,31 +1426,6 @@ static BOOL U_CALLCONV getIcuDataDirectoryUnderWindowsDirectory(char* directoryB
 }
 #endif
 
-#if APPLE_ICU_CHANGES
-// rdar://102831360 (Add Swift Package Manager support to ICU)
-#if defined(USE_PACKAGE_DATA)
-const char* getPackageICUDataPath() {
-    Dl_info dl_info;
-    dladdr(reinterpret_cast<const void*>(getPackageICUDataPath), &dl_info);
-    const char* libraryFilename = dl_info.dli_fname;
-    if (libraryFilename != NULL && libraryFilename[0] != 0) {
-        // Remove the executable name
-        char path[PATH_MAX + 1];
-        strncpy(path, libraryFilename, PATH_MAX);
-        char *lastSlash = strrchr(path, '/');
-        if (lastSlash) {
-            // Terminate the string at /
-            *lastSlash = 0;
-        }
-        // Append the resource bundle path // rdar://104488214
-        strcat(path, "/FoundationICU_FoundationICU.resources");
-        return strdup(path);
-    }
-    return "";
-}
-#endif
-#endif // APPLE_ICU_CHANGES
-
 static void U_CALLCONV dataDirectoryInitFn() {
     /* If we already have the directory, then return immediately. Will happen if user called
      * u_setDataDirectory().
@@ -1459,14 +1434,6 @@ static void U_CALLCONV dataDirectoryInitFn() {
         return;
     }
 
-#if APPLE_ICU_CHANGES
-// rdar://102831360 (Add Swift Package Manager support to ICU)
-#if defined(USE_PACKAGE_DATA)
-    u_setDataDirectory(getPackageICUDataPath());
-    return;
-#endif
-#endif // APPLE_ICU_CHANGES
-
     const char *path = nullptr;
 #if defined(ICU_DATA_DIR_PREFIX_ENV_VAR)
     char datadir_path_buffer[PATH_MAX];
@@ -1603,13 +1570,6 @@ static void U_CALLCONV TimeZoneDataDirInitFn(UErrorCode &status) {
     dir = getenv("ICU_TIMEZONE_FILES_DIR");
 #endif // U_PLATFORM_HAS_WINUWP_API
 
-#if APPLE_ICU_CHANGES
-// rdar://102831360 (Add Swift Package Manager support to ICU)
-#if defined(USE_PACKAGE_DATA)
-    dir = getPackageICUDataPath();
-#endif
-#endif // APPLE_ICU_CHANGES
-
 #if defined(U_TIMEZONE_FILES_DIR)
     if (dir == nullptr) {
         // Build time configuration setting.
-- 
2.39.5 (Apple Git-154)

